docker/Makefile

```diff
@@ -15,11 +18,19 @@ HAIL_PUBLIC_IMAGE = gcr.io/$(PROJECT)/hail-public:$(shell docker images -q --no-
 GENETICS_PUBLIC_LATEST = gcr.io/$(PROJECT)/genetics-public:latest
 GENETICS_PUBLIC_IMAGE = gcr.io/$(PROJECT)/genetics-public:$(shell docker images -q --no-trunc genetics-public:latest | sed -e 's,[^:]*:,,')
 
+.PHONY: hail-ubuntu
+hail-ubuntu: hail-ubuntu-stmp
+
+hail-ubuntu-stmp: hail-ubuntu/Dockerfile hail-ubuntu/hail_pip_install hail-ubuntu/pip.conf
```

Comment:
Can you explain to me why there is this hail-ubuntu-stmp instead of just hail-ubuntu?

Reply:
Cotton chose to create two targets for each image. `image-name` and `image-name-stmp`. I don't know why he didn't use a file named `image-name` to hold the timestamps. Instead, he chose to make `image-name` a PHONY target which just references `image-name-stmp`. The later is a file whose timestamp is updated when an image build is successfully run. File timestamps are used by Make to decide if it needs to re-run a rule.

Reply:
I remember now. He was checking the permissions because if the file permissions were wrong, then it made the builds slower.